demos/noise_suppression_demo/python/noise_suppression_demo.py

```diff
@@ -73,18 +74,25 @@ def main():
 
     log.info('OpenVINO Inference Engine')
     log.info('\tbuild: {}'.format(get_version()))
-    ie = IECore()
+    core = Core()
 
     # read IR
-    log.info("Reading model {}".format(args.model))
-    model_xml = args.model
-    model_bin = model_xml.with_suffix(".bin")
-    ie_encoder = ie.read_network(model=model_xml, weights=model_bin)
+    log.info("Reading model {}".format(args.model_path))
+    model = core.read_model(args.model_path)
 
     # check input and output names
-    input_shapes = {k: v.input_data.shape for k, v in ie_encoder.input_info.items()}
-    input_names = list(ie_encoder.input_info.keys())
-    output_names = list(ie_encoder.outputs.keys())
+    input_shapes = {}
+    input_names = []
+    output_names = []
+    for i in range(len(model.get_parameters())):
```

Comment:
for me, it looks more clear if you separate inputs and outputs checking
and btw we need to guarantee that number of inputs and outputs are equal, before getting info

Reply:
Done